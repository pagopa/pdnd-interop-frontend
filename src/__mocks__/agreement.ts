import { AgreementSummary } from '../../types'

export const agreementActiveEservicePublished: AgreementSummary = {
  id: 'dsfjds-jojoi-jdsfds',
  state: 'ACTIVE',
  eservice: {
    name: 'Riscossione TARI',
    id: 'osdijf-dsjfdisj-jsdfdsj',
    descriptorId: 'lskdfok-jisjdfs-djdsjfn',
    version: '4',
    state: 'PUBLISHED',
  },
  eserviceDescriptorId: 'dnsoifn-dsfjdsiof-dsjfsd',
  consumer: {
    name: 'Comune di Milano',
    id: 'afd_dss_cds',
  },
  producer: {
    name: 'Comune di Bologna',
    id: 'lds_kji_dsj',
  },
  attributes: [],
  suspendedByProducer: false,
  suspendedBySubscriber: false,
}

export const agreementSuspendedByProducerEservicePublished: AgreementSummary = {
  id: 'dsfjds-jojoi-jdsfds',
  state: 'SUSPENDED',
  eservice: {
    name: 'Riscossione TARI',
    id: 'osdijf-dsjfdisj-jsdfdsj',
    descriptorId: 'lskdfok-jisjdfs-djdsjfn',
    version: '4',
    state: 'PUBLISHED',
  },
  eserviceDescriptorId: 'dnsoifn-dsfjdsiof-dsjfsd',
  consumer: {
    name: 'Comune di Milano',
    id: 'afd_dss_cds',
  },
  producer: {
    name: 'Comune di Bologna',
    id: 'lds_kji_dsj',
  },
  attributes: [],
  suspendedByProducer: true,
  suspendedBySubscriber: false,
}

export const agreementSuspendedBySubscriberEservicePublished: AgreementSummary = {
  id: 'dsfjds-jojoi-jdsfds',
  state: 'SUSPENDED',
  eservice: {
    name: 'Riscossione TARI',
    id: 'osdijf-dsjfdisj-jsdfdsj',
    descriptorId: 'lskdfok-jisjdfs-djdsjfn',
    version: '4',
    state: 'PUBLISHED',
  },
  eserviceDescriptorId: 'dnsoifn-dsfjdsiof-dsjfsd',
  consumer: {
    name: 'Comune di Milano',
    id: 'afd_dss_cds',
  },
  producer: {
    name: 'Comune di Bologna',
    id: 'lds_kji_dsj',
  },
  attributes: [],
  suspendedByProducer: false,
  suspendedBySubscriber: true,
}
