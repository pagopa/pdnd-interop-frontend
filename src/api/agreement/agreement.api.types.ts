import type { AgreementState } from '@/types/agreement.types'
import type { PaginationParams } from '../api.types'

export type GetListAgreementQueryFilters = {
  producersIds?: Array<string>
  consumersIds?: Array<string>
  states?: Array<AgreementState>
  eservicesIds?: Array<string>
  showOnlyUpgradeable?: boolean
}

export type GetListAgreementQueryParams = GetListAgreementQueryFilters & PaginationParams

export type UploadAgreementDraftDocumentPayload = {
  name: string
  prettyName: string
  doc: File
}
