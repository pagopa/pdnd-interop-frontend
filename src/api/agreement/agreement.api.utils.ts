import type { AgreementSummary } from '@/types/agreement.types'

export function updateAgreementsListCache(
  newAgreementData: AgreementSummary,
  agreementsListCache: Array<AgreementSummary> = []
) {
  const agreementsListCacheCopy = [...agreementsListCache]
  const index = agreementsListCacheCopy.findIndex(
    (agreement) => agreement.id === newAgreementData.id
  )
  if (index !== -1) {
    agreementsListCacheCopy[index] = newAgreementData
  }
  return agreementsListCacheCopy
}

export function removeAgreementFromListCache(
  agreementId: string,
  agreementsListCache: Array<AgreementSummary> = []
) {
  return agreementsListCache.filter((agreementCache) => agreementCache.id !== agreementId)
}
