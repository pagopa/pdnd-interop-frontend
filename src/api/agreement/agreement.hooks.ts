import { useQueryClient } from '@tanstack/react-query'
import { useTranslation } from 'react-i18next'
import { useMutationWrapper, useQueryWrapper } from '../react-query-wrappers'
import type { UseQueryWrapperOptions } from '../react-query-wrappers/react-query-wrappers.types'
import { useDownloadFile } from '../react-query-wrappers/useDownloadFile'
import AgreementServices from './agreement.services'
import type {
  Agreement,
  Agreements,
  CompactEServicesLight,
  CompactOrganizations,
  GetAgreementConsumersParams,
  GetAgreementEServiceConsumersParams,
  GetAgreementEServiceProducersParams,
  GetAgreementProducersParams,
  GetAgreementsParams,
} from '../api.generatedTypes'

export enum AgreementQueryKeys {
  GetList = 'AgreementGetList',
  GetSingle = 'AgreementGetSingle',
  GetProducers = 'AgreementGetProducers',
  GetConsumers = 'AgreementGetConsumers',
  GetProducerEServiceList = 'AgreementGetProducerEServiceList',
  GetConsumerEServiceList = 'AgreementGetConsumerEServiceList',
}

function useGetList(params: GetAgreementsParams, config: UseQueryWrapperOptions<Agreements>) {
  return useQueryWrapper(
    [AgreementQueryKeys.GetList, params],
    () => AgreementServices.getList(params),
    config
  )
}

function useGetSingle(agreementId: string, config?: UseQueryWrapperOptions<Agreement>) {
  return useQueryWrapper(
    [AgreementQueryKeys.GetSingle, agreementId],
    () => AgreementServices.getSingle(agreementId),
    config
  )
}

function useGetProducers(
  params: GetAgreementProducersParams,
  config?: UseQueryWrapperOptions<CompactOrganizations>
) {
  return useQueryWrapper(
    [AgreementQueryKeys.GetProducers, params],
    () => AgreementServices.getProducers(params),
    config
  )
}

function useGetConsumers(
  params: GetAgreementConsumersParams,
  config?: UseQueryWrapperOptions<CompactOrganizations>
) {
  return useQueryWrapper(
    [AgreementQueryKeys.GetConsumers, params],
    () => AgreementServices.getConsumers(params),
    config
  )
}

function usePrefetchSingle() {
  const queryClient = useQueryClient()
  return (agreementId: string) =>
    queryClient.prefetchQuery([AgreementQueryKeys.GetSingle, agreementId], () =>
      AgreementServices.getSingle(agreementId)
    )
}

function useGetProducerEServiceList(
  params: GetAgreementEServiceProducersParams,
  config: UseQueryWrapperOptions<CompactEServicesLight>
) {
  return useQueryWrapper(
    [AgreementQueryKeys.GetProducerEServiceList, params],
    () => AgreementServices.getProducerEServiceList(params),
    config
  )
}

function useGetConsumerEServiceList(
  params: GetAgreementEServiceConsumersParams,
  config: UseQueryWrapperOptions<CompactEServicesLight>
) {
  return useQueryWrapper(
    [AgreementQueryKeys.GetConsumerEServiceList, params],
    () => AgreementServices.getConsumerEServiceList(params),
    config
  )
}

function useCreateDraft() {
  const { t } = useTranslation('mutations-feedback', { keyPrefix: 'agreement.createDraft' })
  return useMutationWrapper(AgreementServices.createDraft, {
    suppressSuccessToast: true,
    errorToastLabel: t('outcome.error'),
    loadingLabel: t('loading'),
    showConfirmationDialog: true,
    dialogConfig: {
      title: t('confirmDialog.title'),
      description: ({ eserviceName, eserviceVersion }) =>
        t('confirmDialog.description', { name: eserviceName, version: eserviceVersion }),
      proceedLabel: t('confirmDialog.proceedLabel'),
    },
  })
}

function useSubmitDraft() {
  const { t } = useTranslation('mutations-feedback', { keyPrefix: 'agreement.submitDraft' })
  return useMutationWrapper(AgreementServices.submitDraft, {
    successToastLabel: t('outcome.success'),
    errorToastLabel: t('outcome.error'),
    loadingLabel: t('loading'),
    showConfirmationDialog: true,
    dialogConfig: {
      title: t('confirmDialog.title'),
      description: t('confirmDialog.description'),
    },
  })
}

function useDeleteDraft() {
  const { t } = useTranslation('mutations-feedback', { keyPrefix: 'agreement.deleteDraft' })
  return useMutationWrapper(AgreementServices.deleteDraft, {
    successToastLabel: t('outcome.success'),
    errorToastLabel: t('outcome.error'),
    loadingLabel: t('loading'),
    showConfirmationDialog: true,
    dialogConfig: {
      title: t('confirmDialog.title'),
      description: t('confirmDialog.description'),
    },
  })
}

function useUpdateDraft() {
  const { t } = useTranslation('mutations-feedback', { keyPrefix: 'agreement.updateDraft' })
  return useMutationWrapper(AgreementServices.updateDraft, {
    successToastLabel: t('outcome.success'),
    errorToastLabel: t('outcome.error'),
    loadingLabel: t('loading'),
  })
}

function useDownloadDocument() {
  const { t } = useTranslation('mutations-feedback', {
    keyPrefix: 'agreement.downloadDraftDocument',
  })
  return useDownloadFile(AgreementServices.downloadDraftDocument, {
    errorToastLabel: t('outcome.error'),
    loadingLabel: t('loading'),
  })
}

function useUploadDraftDocument() {
  const { t } = useTranslation('mutations-feedback', {
    keyPrefix: 'agreement.uploadDraftDocument',
  })
  return useMutationWrapper(AgreementServices.uploadDraftDocument, {
    suppressSuccessToast: true,
    errorToastLabel: t('outcome.error'),
    loadingLabel: t('loading'),
  })
}

function useDeleteDraftDocument() {
  const { t } = useTranslation('mutations-feedback', {
    keyPrefix: 'agreement.deleteDraftDocument',
  })
  return useMutationWrapper(AgreementServices.deleteDraftDocument, {
    successToastLabel: t('outcome.success'),
    errorToastLabel: t('outcome.error'),
    loadingLabel: t('loading'),
  })
}

function useActivate() {
  const { t } = useTranslation('mutations-feedback', { keyPrefix: 'agreement.activate' })
  return useMutationWrapper(AgreementServices.activate, {
    successToastLabel: t('outcome.success'),
    errorToastLabel: t('outcome.error'),
    loadingLabel: t('loading'),
    showConfirmationDialog: true,
    dialogConfig: {
      title: t('confirmDialog.title'),
      description: t('confirmDialog.description'),
    },
  })
}

function useReject() {
  const { t } = useTranslation('mutations-feedback', { keyPrefix: 'agreement.reject' })
  return useMutationWrapper(AgreementServices.reject, {
    suppressSuccessToast: true,
    errorToastLabel: t('outcome.error'),
    loadingLabel: t('loading'),
  })
}

function useClone() {
  const { t } = useTranslation('mutations-feedback', { keyPrefix: 'agreement.clone' })
  return useMutationWrapper(AgreementServices.clone, {
    successToastLabel: t('outcome.success'),
    errorToastLabel: t('outcome.error'),
    loadingLabel: t('loading'),
    showConfirmationDialog: true,
    dialogConfig: {
      title: t('confirmDialog.title'),
      description: t('confirmDialog.description'),
    },
  })
}

function useSuspend() {
  const { t } = useTranslation('mutations-feedback', { keyPrefix: 'agreement.suspend' })
  return useMutationWrapper(AgreementServices.suspend, {
    successToastLabel: t('outcome.success'),
    errorToastLabel: t('outcome.error'),
    loadingLabel: t('loading'),
    showConfirmationDialog: true,
    dialogConfig: {
      title: t('confirmDialog.title'),
      description: t('confirmDialog.description'),
    },
  })
}

function useUpgrade() {
  const { t } = useTranslation('mutations-feedback', { keyPrefix: 'agreement.upgrade' })
  return useMutationWrapper(AgreementServices.upgrade, {
    successToastLabel: t('outcome.success'),
    errorToastLabel: t('outcome.error'),
    loadingLabel: t('loading'),
    showConfirmationDialog: true,
    dialogConfig: {
      title: t('confirmDialog.title'),
      description: t('confirmDialog.description'),
    },
  })
}

function useDownloadContract() {
  const { t } = useTranslation('mutations-feedback', {
    keyPrefix: 'agreement.downloadContract',
  })
  return useDownloadFile(AgreementServices.downloadContract, {
    errorToastLabel: t('outcome.error'),
    loadingLabel: t('loading'),
  })
}

export const AgreementQueries = {
  useGetList,
  useGetSingle,
  usePrefetchSingle,
  useGetProducers,
  useGetConsumers,
  useGetProducerEServiceList,
  useGetConsumerEServiceList,
}

export const AgreementMutations = {
  useCreateDraft,
  useSubmitDraft,
  useDeleteDraft,
  useUpdateDraft,
  useUploadDraftDocument,
  useDeleteDraftDocument,
  useActivate,
  useReject,
  useSuspend,
  useUpgrade,
  useClone,
}

export const AgreementDownloads = {
  useDownloadDocument,
  useDownloadContract,
}
