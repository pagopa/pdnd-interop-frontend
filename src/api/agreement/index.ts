export * from './agreement.mutations'
export * from './agreement.queries'
export * from './agreement.downloads'
export * from './agreement.services'
