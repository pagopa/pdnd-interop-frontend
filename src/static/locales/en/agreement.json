{
  "loadingSingleLabel": "Loading request",
  "loadingMultiLabel": "Loading requests",
  "backToRequestsBtn": "Go back to requests",
  "read": {
    "title": "Manage fruition request",
    "noPurposeAlert": "Your agreement request is active. <1>Create your first purpose</1>",
    "updateGuide": {
      "alertLabel": "A new version of <1>{{eserviceName}}</1> e-service is available. Before updating, read the guide below",
      "title": "Upgrade guide",
      "description": "It is possible to update the fruition request to the latest version of <1>{{eserviceName}}</1> e-service. Here are some useful informations",
      "upgradeGuideLinkLabel": "Upgrade fuide",
      "linksLabel": "Useful links",
      "eserviceLinkLabel": "E-service card <1>{{eserviceName}}, version {{version}}</1>",
      "upgradeBtn": "Upgrade"
    },
    "generalInformations": {
      "title": "General Informations",
      "eserviceField": {
        "label": "Request for",
        "goToEServiceBtn": "Go to e-service",
        "versionLabel": "version"
      },
      "providerField": {
        "label": "Provider",
        "infoLabel": "Max 1000 chars"
      },
      "consumerField": {
        "label": "Consumer"
      },
      "requestStatusField": {
        "label": "Request status"
      },
      "printableCopyField": {
        "label": "Printable copy",
        "docLabel": "Fruition request"
      }
    },
    "consumerMessage": {
      "title": "Message for provider"
    },
    "rejectionMessage": {
      "title": "Fruition request rejection reason",
      "description": "The provider states their reason for refusing to activate this fruition request. You can always submit a new one"
    },
    "attributes": {
      "howLink": "How?",
      "certified": {
        "title": "Certified attributes",
        "subtitle": "Questi attributi sono certificati da una fonte autoritativa riconosciuta, e non necessitano di ulteriori verifiche.",
        "emptyLabel": "There are no required certified attributes for this e-service"
      },
      "verified": {
        "title": "Verified attributes",
        "subtitle": "Questi attributi necessitano di verifica da parte di almeno un erogatore. Il fruitore potrebbe già possederne alcuni (o tutti).",
        "emptyLabel": "There are no required verified attributes for this e-service"
      },
      "declared": {
        "title": "Declared attributes",
        "subtitle": "Il fruitore deve dichiarare sotto la propria responsabilità di possedere questi attributi; potrebbe già possederne alcuni (o tutti).",
        "emptyLabel": "There are no required declared attributes for this e-service"
      }
    }
  },
  "edit": {
    "title": "Manage fruition request",
    "generalInformations": {
      "title": "General Informations",
      "eserviceField": {
        "label": "Request for",
        "goToEServiceBtn": "Go to e-service",
        "versionLabel": "version"
      },
      "providerField": {
        "label": "Provider"
      },
      "requestStatusField": {
        "label": "Request status"
      }
    },
    "attribute": {
      "subtitle": "You must own:",
      "certified": {
        "description": "These attributes are certified by a recognized authoritative source, and do not require further certification."
      },
      "verified": {
        "description": "These attributes require verification by at least one provider. You may already own some (or all) of them."
      },
      "declared": {
        "description": "You must declare under your own responsibility that you possess these attributes."
      }
    },
    "consumerNotes": {
      "title": "Message for provider",
      "description": "If there are any groups of attributes marked with “or”, indicate in this message which of those you want to be recognized, based on the documentation you have uploaded. Enter any other useful notes",
      "field": {
        "label": "Message",
        "infoLabel": "Max 1000 chars"
      }
    },
    "documents": {
      "title": "Documentation",
      "description": "Enter all the documents necessary for the recognition of your attributes",
      "uploadBtn": "Upload",
      "documentPrettynameField": {
        "label": "Document name",
        "infoLabel": "Min 5 chars, max 60 chars"
      }
    },
    "bottomPageActionCard": {
      "title": "Quick publish",
      "description": "Have you entered all the information for this agreement request? From here you can immediately publish a draft, or delete it. If you want to post later, just save the draft above or leave this page",
      "cancelBtn": "Delete draft",
      "submitBtn": "Publish agreement request",
      "updateBtn": "Save draft"
    },
    "backToRequestsBtn": "Go back to requests"
  },
  "list": {
    "eserviceName": "{{name}}, version {{version}}",
    "filters": {
      "eserviceField": {
        "label": "Find by e-service"
      },
      "statusField": {
        "label": "Agreement status",
        "optionLabels": {
          "ACTIVE": "Active",
          "DRAFT": "Draft",
          "SUSPENDED": "Suspended",
          "PENDING": "Pending",
          "ARCHIVED": "Archived",
          "REJECTED": "Rejected",
          "MISSING_CERTIFIED_ATTRIBUTES": "Missing certified attributes"
        }
      },
      "consumerField": {
        "label": "Find by consumer"
      },
      "providerField": {
        "label": "Find by provider"
      }
    },
    "provider": {
      "title": "Fruition requests",
      "description": "In this section, you can manage all the fruition requests you have received from parties that want to subscribe to your e-services"
    },
    "subscriber": {
      "title": "Your requests",
      "description": "In this section, you can manage all the fruition requests you have sent to e-service Providers"
    }
  }
}
