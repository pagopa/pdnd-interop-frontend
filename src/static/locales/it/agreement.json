{
  "loadingSingleLabel": "Stiamo caricando la richiesta",
  "loadingMultiLabel": "Stiamo caricando la lista di richieste",
  "backToRequestsBtn": "Torna alle richieste",
  "read": {
    "title": "Gestisci richiesta di fruizione",
    "noPurposeAlert": "La richiesta di fruizione è attiva. <1>Crea la tua prima finalità</1>",
    "updateGuide": {
      "alertLabel": "È disponibile una nuova versione dell’e-service <1>{{eserviceName}}</1>. Prima di effettuare l’aggiornamento, leggi la guida di seguito",
      "title": "Guida all’aggiornamento",
      "description": "È possibile aggiornare la richiesta di fruizione all’ultima versione dell’e-service <1>{{eserviceName}}</1>. Ecco alcune indicazioni utili",
      "upgradeGuideLinkLabel": "Guida all'aggiornamento",
      "linksLabel": "Link utili",
      "eserviceLinkLabel": "Scheda e-service <1>{{eserviceName}}, versione {{version}}</1>",
      "upgradeBtn": "Aggiorna"
    },
    "generalInformations": {
      "title": "Informazioni generali",
      "eserviceField": {
        "label": "Richiesta relativa a",
        "goToEServiceBtn": "Vedi e-service",
        "versionLabel": "versione"
      },
      "providerField": {
        "label": "Erogatore"
      },
      "consumerField": {
        "label": "Fruitore"
      },
      "requestStatusField": {
        "label": "Stato della richiesta"
      },
      "printableCopyField": {
        "label": "Copia stampabile",
        "docLabel": "Richiesta di fruizione"
      }
    },
    "consumerMessage": {
      "title": "Messaggio per l’erogatore"
    },
    "rejectionMessage": {
      "title": "Motivazione rifiuto richiesta di fruizione",
      "description": "L'erogatore ha motivato così il rifiuto di attivare questa richiesta di fruizione. Il fruitore potrà sempre presentarne una nuova"
    },
    "attributes": {
      "howLink": "Come funziona?",
      "certified": {
        "title": "Attributi certificati",
        "subtitle": "Questi attributi sono certificati da una fonte autoritativa riconosciuta, e non necessitano di ulteriori verifiche.",
        "emptyLabel": "Non ci sono attributi certificati richiesti per questo e-service"
      },
      "verified": {
        "title": "Attributi verificati",
        "subtitle": "Questi attributi necessitano di verifica da parte di almeno un erogatore. Il fruitore potrebbe già possederne alcuni (o tutti).",
        "emptyLabel": "Non ci sono attributi verificati richiesti per questo e-service"
      },
      "declared": {
        "title": "Attributi dichiarati",
        "subtitle": "Il fruitore deve dichiarare sotto la propria responsabilità di possedere questi attributi; potrebbe già possederne alcuni (o tutti).",
        "emptyLabel": "Non ci sono attributi dichiarati richiesti per questo e-service"
      }
    }
  },
  "edit": {
    "title": "Gestisci richiesta di fruizione",
    "generalInformations": {
      "title": "Informazioni generali",
      "eserviceField": {
        "label": "Richiesta relativa a",
        "goToEServiceBtn": "Vedi e-service",
        "versionLabel": "versione"
      },
      "providerField": {
        "label": "Erogatore"
      },
      "requestStatusField": {
        "label": "Stato della richiesta"
      }
    },
    "attribute": {
      "subtitle": "Devi possedere:",
      "certified": {
        "description": "Questi attributi sono certificati da una fonte autoritativa riconosciuta, e non necessitano di ulteriori verifiche."
      },
      "verified": {
        "description": "Questi attributi necessitano verifica da parte di almeno un erogatore. Il fruitore potrebbe già possederne alcuni (o tutti)."
      },
      "declared": {
        "description": "Il fruitore deve dichiarare sotto la propria responsabilità di possedere questi attributi; potrebbe già possederne alcuni (o tutti)."
      }
    },
    "consumerNotes": {
      "title": "messaggio per l’erogatore",
      "description": "Se ci sono gruppi di attributi contraddistinti da “oppure”, indica in questo messaggio quale tra quelli ti aspetti venga riconosciuto, in base alla documentazione che hai caricato. Inserisci eventuali altre note utili",
      "field": {
        "label": "Messaggio",
        "infoLabel": "Max 1000 caratteri"
      }
    },
    "documents": {
      "title": "Documentazione",
      "description": "Inserisci tutti i documenti necessari al riconoscimento dei tuoi attributi",
      "uploadBtn": "Carica",
      "documentPrettynameField": {
        "label": "Nome documento",
        "infoLabel": "Min 5 caratteri, max 60 caratteri"
      }
    },
    "bottomPageActionCard": {
      "title": "Azioni rapide di pubblicazione",
      "description": "Hai inserito tutte le informazioni per questa richiesta di fruizione? Da qui puoi pubblicare immediatamente una bozza, oppure cancellarla. Se desideri pubblicare più tardi, salva solo la bozza sopra o abbandona questa pagina",
      "cancelBtn": "Cancella bozza",
      "submitBtn": "Inoltra richiesta di fruizione",
      "updateBtn": "Salva bozza"
    }
  },
  "list": {
    "eserviceName": "{{name}}, v. {{version}}",
    "filters": {
      "eserviceField": {
        "label": "Cerca per e-service"
      },
      "statusField": {
        "label": "Stato della richiesta",
        "optionLabels": {
          "ACTIVE": "Attivo",
          "DRAFT": "In bozza",
          "SUSPENDED": "Sospeso",
          "PENDING": "In attesa di approvazione",
          "ARCHIVED": "Archiviato",
          "REJECTED": "Rifiutato",
          "MISSING_CERTIFIED_ATTRIBUTES": "Attributi certificati mancanti"
        }
      },
      "consumerField": {
        "label": "Cerca per fruitore"
      },
      "providerField": {
        "label": "Cerca per erogatore"
      }
    },
    "provider": {
      "title": "Richieste di fruizione",
      "description": "In quest'area puoi gestire tutte le richieste di fruizione che ti sono state inoltrate da enti che intendono fruire dei tuoi e-service"
    },
    "subscriber": {
      "title": "Le tue richieste",
      "description": "In quest'area puoi gestire tutte le richieste di fruizione che hai sottoscritto presso Enti erogatori"
    }
  }
}
