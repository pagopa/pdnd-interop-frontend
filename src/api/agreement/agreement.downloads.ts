import { useTranslation } from 'react-i18next'
import { useDownloadFile } from '../hooks'
import { AgreementServices } from './agreement.services'

function useDownloadDocument() {
  const { t } = useTranslation('mutations-feedback', {
    keyPrefix: 'agreement.downloadDraftDocument',
  })
  return useDownloadFile(AgreementServices.downloadDraftDocument, {
    errorToastLabel: t('outcome.error'),
    loadingLabel: t('loading'),
  })
}

function useDownloadContract() {
  const { t } = useTranslation('mutations-feedback', {
    keyPrefix: 'agreement.downloadContract',
  })
  return useDownloadFile(AgreementServices.downloadContract, {
    errorToastLabel: t('outcome.error'),
    loadingLabel: t('loading'),
  })
}

export const AgreementDownloads = {
  useDownloadDocument,
  useDownloadContract,
}
