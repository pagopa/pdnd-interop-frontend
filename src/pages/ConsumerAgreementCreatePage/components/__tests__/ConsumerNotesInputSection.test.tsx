import React from 'react'
import { render } from '@testing-library/react'
import {
  ConsumerNotesInputSection,
  ConsumerNotesInputSectionSkeleton,
} from '../ConsumerNotesInputSection'
import userEvent from '@testing-library/user-event'
import { vi } from 'vitest'
import { createMockAgreement } from '__mocks__/data/agreement.mocks'

const commonProps = {
  setConsumerNotes: vi.fn(),
  consumerNotes: '',
}

describe('ConsumerNotesInputSection', () => {
  it('should match snapshot', () => {
    const { baseElement } = render(<ConsumerNotesInputSection {...commonProps} />)
    expect(baseElement).toMatchSnapshot()
  })

  it('should correctly call the setConsumerNotes function', async () => {
    const setConsumerNotesFn = vi.fn()
    const screen = render(
      <ConsumerNotesInputSection
        {...commonProps}
        setConsumerNotes={setConsumerNotesFn}
        agreement={createMockAgreement({ consumerNotes: '' })}
      />
    )
    const user = userEvent.setup()
    const consumerNotesInput = screen.getByLabelText('field.label')
    await user.type(consumerNotesInput, 'a')
    expect(setConsumerNotesFn).toHaveBeenCalledWith('a')
  })

  it('should sync with the agreement consumerNotes if it changes', () => {
    const setConsumerNotesFn = vi.fn()

    const screen = render(
      <ConsumerNotesInputSection
        {...commonProps}
        setConsumerNotes={setConsumerNotesFn}
        agreement={createMockAgreement({ consumerNotes: 'foo' })}
      />
    )

    screen.rerender(
      <ConsumerNotesInputSection
        {...commonProps}
        setConsumerNotes={setConsumerNotesFn}
        agreement={createMockAgreement({ consumerNotes: 'bar' })}
      />
    )

    expect(setConsumerNotesFn).toHaveBeenCalledWith('bar')
  })

  // it('should correctly validate the consumer notes field user input', async () => {
  //   const screen = render(
  //     <ConsumerNotesInputSection
  //       {...commonProps}
  //       agreement={createMockAgreement({ consumerNotes: '' })}
  //     />
  //   )

  //   const user = userEvent.setup()
  //   const consumerNotesInput = screen.getByLabelText('Consumer notes')
  //   user.type(consumerNotesInput, 'a'.repeat(1001))
  //   expect(consumerNotesInput).toHaveValue('a'.repeat(1000))
  // })
})

describe('ConsumerNotestInputSectionSkeleton', () => {
  it('should match snapshot', () => {
    const { baseElement } = render(<ConsumerNotesInputSectionSkeleton />)
    expect(baseElement).toMatchSnapshot()
  })
})
