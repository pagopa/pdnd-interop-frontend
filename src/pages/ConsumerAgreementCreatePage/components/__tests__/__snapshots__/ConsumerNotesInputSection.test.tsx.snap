// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ConsumerNotesInputSection > should match snapshot 1`] = `
<body>
  <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1w5x330-MuiPaper-root"
    >
      <div
        class="css-1okj3ks-MuiStack-root"
      >
        <h2
          class="MuiTypography-root MuiTypography-overline css-b3fch5-MuiTypography-root"
        >
          title
        </h2>
        <span
          class="MuiTypography-root MuiTypography-caption css-ol7wxs-MuiTypography-root"
        >
          description
        </span>
      </div>
      <div
        class="MuiBox-root css-1yuhvjn"
      >
        <div
          class="MuiBox-root css-1e1z9m8"
        >
          <div
            class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              for=":r0:"
              id=":r0:-label"
            >
              field.label
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-multiline css-dpjnhs-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <textarea
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r0:"
                maxlength="1000"
                name="consumerNotes"
                rows="6"
                style="height: 0px; overflow: hidden;"
              />
              <textarea
                aria-hidden="true"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                readonly=""
                style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding: 0px; width: 100%;"
                tabindex="-1"
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-14lo706"
                >
                  <span>
                    field.label
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <p
            class="MuiTypography-root MuiTypography-caption css-6n3pnq-MuiTypography-root"
          >
            field.infoLabel
          </p>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`ConsumerNotestInputSectionSkeleton > should match snapshot 1`] = `
<body>
  <div>
    <span
      class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-4i8vzz-MuiSkeleton-root"
      style="height: 326px;"
    />
  </div>
</body>
`;
